### PR TITLE
feat: Buy Now

### PR DESCRIPTION
**Pull request by FathimaSayeeda**

### CODE REVIEW

File: vestido/vestido-storefront/modules/ProductListView/AddToCartButton.tsx [hunk 1]
```diff
@@ -38,15 +43,40 @@ const AddToCartButton: React.FC<AddToCartButtonProps> = ({
...
+              <div>
+                <p className="font-semibold">Product Name</p>
+                <p className="text-sm text-gray-500">
+                  This is a short description of the product.
+                </p>
+              </div>
+            </div>
+          ),
+        });
+        console.log('passed toast');
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/ProductListView/AddToCartButton.tsx:79]: Why is this here?

File: vestido/vestido-storefront/modules/products/product-view.tsx [hunk 2]
```diff
@@ -62,6 +64,18 @@ const ProductView: React.FC<ProductViewProps> = ({ itemId }) => {
...
+  const handleBuyNow = () => {
+    if (!isAuthenticated) {
+      routeToLogin();
+      return;
+    }
+
+    if (item) {
+      router.replace(
+        `/checkout?buyNowItemId=${item.id}&buyNowVariantId=${selectedVariantId}`,
+      );
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/products/product-view.tsx:76]: Router.push please. Please read upon replace() vs push()

File: vestido/vestido-storefront/modules/products/product-view.tsx [hunk 3]
```diff
@@ -128,32 +142,35 @@ const ProductView: React.FC<ProductViewProps> = ({ itemId }) => {
...
-              >
-                BUY NOW
-              </Button>
-            </div> */}
-            <AddToWishListButton
-              itemId={item?.id || ''}
-              className="border border-2 border-[#48CAB2] font-medium text-xs  h-full self-center p-4"
-            />
+            <div className="flex flex-1 gap-2">
+              {' '}
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/products/product-view.tsx:160]: Please remove this.

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx [hunk 4]
```diff
@@ -79,15 +86,32 @@ const CheckoutView: React.FC = () => {
...
+            {
+              itemId: buyNowItemId ?? '',
+              qty: 1,
+              price: buyNowItem.price,
+              variantId: buyNowVariantId ?? '',
+            },
+          ],
+          paymentType: data.paymentType,
+        }
+      : data;
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:111]: Form already has values from `checkoutItems` so this is unnecessary?

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx [hunk 5]
```diff
@@ -79,15 +86,32 @@ const CheckoutView: React.FC = () => {
...
-    cartItems?.data.reduce((total, item) => {
-      return total + item.qty * item.item.price;
-    }, 0) ?? 0;
+  const totalPrice = buyNowItem
+    ? (checkoutItems?.reduce((total, item) => {
+        return total + item.item.price;
+      }, 0) ?? 0)
+    : (cartItems?.reduce((total, item) => {
+        return total + item.qty * item.item.price;
+      }, 0) ?? 0);
```
Comment (fahimalizain) [inline vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:95]: You only have to do this with `checkoutItems`?

**Review by fahimalizain — CHANGES_REQUESTED** [review]
In CheckoutView.tsx, dealing buyNowItem vs CartItems is being overdone.
`checkoutItems` = `[buyNowItem]` || `cartIitems`.

This is the only line where the decision should be made. Everywhere else it should just refer `checkoutItems`.

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx (hunk 4 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:111]: ` useEffect(() => {
    if (cartItems?.length) {
      form.setValue(
        'orderItems',
        cartItems?.map((cartItem) => ({
          itemId: cartItem.itemId,
          price: cartItem.item.price,
          qty: cartItem.qty,
          variantId: cartItem.variantId ?? null,
        })),
      );
    }
  }, [cartItems, form]);`
  
  this should be changed to checkoutItems ??

Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:111]: but then what about qty or variantID ?

File: vestido/vestido-storefront/modules/products/product-view.tsx (hunk 3 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/products/product-view.tsx:160]: {' '}   --- this ??

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx (hunk 4 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:111]: image
first, i should add a constVariantId???

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx (hunk 5 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx:95]: ok

Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx]: this is done

File: vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx (hunk 4 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/Checkout/CheckoutView.tsx]: this is done

File: vestido/vestido-storefront/modules/products/product-view.tsx (hunk 3 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/products/product-view.tsx]: done

File: vestido/vestido-storefront/modules/products/product-view.tsx (hunk 2 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/products/product-view.tsx]: done

File: vestido/vestido-storefront/modules/ProductListView/AddToCartButton.tsx (hunk 1 above)
Comment (FathimaSayeeda) [reply vestido/vestido-storefront/modules/ProductListView/AddToCartButton.tsx]: done

**Review by fahimalizain — APPROVED** [review]
